File: 6/Python/main.py
#!/usr/bin/python

import time

print("This program finds the difference between the sum of squares of the")
print("first n natural numbers and the square of the sum of the first n")
print("natural numbers")

n = int(raw_input("Enter n > "))

start = time.time()

# Program expands the square of the sum of the first n natural numbers.
# So this expression is in the form (a+b+c...)^2 = a^2+2ab+2bc+b^2...
# So all the x^2 values will cancel out when subtracted.
sum = 0
for i in range(1, n+1):
    for j in range(1, n+1):
        if i != j:
            sum += i * j
end = time.time()

print("Result: {}".format(sum))
print("{} s".format(end - start))

File: 6/Haskell/main.hs
{-
 - Author: Matias Grioni
 - Created: 7/7/15
 -
 - Finds the difference between the square of the sum and the sum of the
 - squares of numbers between 1 to n.
 -}

module Main where

main = do
    putStrLn "Finds the difference between the square of the sum and the sum"
    putStrLn "of the squares fo numbers between 1 to n"
    putStrLn "Enter n > "
    input <- getLine
    let n = read input :: Int

    print $ (sum [1..n]) ^ 2 - (sum $ map (^2) [1..n])
